### PR TITLE
TF-4520 Team mailbox folder should be sorted correctly

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#4520

Teamc mailbox sub-folders are sorted purely alphabetically, causing system folders (Inbox, Drafts, Sent, Trash, etc.) to appear mixed in with user-created folders instead of being grouped at the top in canonical order.

## Root cause

`TreeBuilder` applied `sortByMailboxNameNodeChildren` uniformly across all mailbox trees. There was no distinction between team mailbox root nodes and regular nodes, so system folders received no special ordering — they were sorted by display name alongside user-created folders.

## Solution

- Added `_sortWithSystemFoldersFirst`: sorts children of a team mailbox root by placing system folders first (in canonical role order: Inbox → Drafts → Outbox → Sent → Trash → Spam/Junk → Templates → Archive), then user folders alphabetically. System folder identity is determined by **role**, not display name, so localised folder names sort correctly.

- Added `_sortTeamMailboxChildrenAlphabetically`: recursively walks the team mailbox subtree, applying system-first ordering at team root level and alphabetical ordering at all deeper levels.

- Extracted `_placeNodeInTree` and `_finalizeTrees` to eliminate the duplicated parent-resolution and post-build sort logic that existed across both build paths (`generateMailboxTreeInUI` / `generateMailboxTreeInUIAfterRefreshChanges`).

- Default and personal trees are unaffected.

## Resolved

Screenshot 2026-05-27 at 18 08 35

## Summary by CodeRabbit

* **New Features**
  * Mailbox tree rendering improved: team mailboxes show system folders first in a canonical order, then user folders alphabetically; default and personal trees remain alphabetized. Child folders inherit parent deactivation state when added.

* **Refactor**
  * Streamlined tree-building and centralized deterministic child ordering/finalization.
  * Consolidated system-folder display names and icons into lookup-driven helpers.

* **Tests**
  * Expanded tests covering team sorting, system-first behavior, edge cases, and nested alphabetical ordering.

[Review Change Stack](https://app.coderabbit.ai/change-stack/linagora/tmail-flutter/pull/4562?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lib/features/mailbox/presentation/model/mailbox_tree_builder.dart [hunk 1]
```diff
@@ -162,6 +195,48 @@ class TreeBuilder {
...
+      if (aIsSystem) return -1;
+      if (bIsSystem) return 1;
+      return a.item.name?.compareAlphabetically(b.item.name) ?? -1;
+    });
+  }
+
+  void _applyTeamMailboxSorting(MailboxNode node) {
+    if (node.childrenItems == null || node.childrenItems!.isEmpty) return;
+    if (node.item.isTeamMailboxes) {
+      _sortWithSystemFoldersFirst(node);
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/model/mailbox_tree_builder.dart:219]: right now, in PROD, all mailbox in team-mailbox does not have `Role`, so the `drafts` above `inbox`

Image

File: lib/features/mailbox/presentation/model/mailbox_tree_builder.dart
Comment (hoangdat) [inline lib/features/mailbox/presentation/model/mailbox_tree_builder.dart:1]: `generateMailboxTree` not in use

File: lib/features/mailbox/presentation/model/mailbox_tree_builder.dart (hunk 1 above)
Comment (dab246) [reply lib/features/mailbox/presentation/model/mailbox_tree_builder.dart:219]: Fixed

Screenshot 2026-05-27 at 18 08 35

File: lib/features/mailbox/presentation/model/mailbox_tree_builder.dart
Comment (dab246) [reply lib/features/mailbox/presentation/model/mailbox_tree_builder.dart:1]: removed

**Review by hoangdat — APPROVED** [review]
